### PR TITLE
NPCs - added a new hunter NPC that trades with the player to two locations

### PR DESCRIPTION
**Pull request by Brytenwalda**
#### Summary
 ```SUMMARY: Content "A new type of a hunter NPC will spawn in some cabins and campsites"```

#### Purpose of change
More NPC hunters would make the world more plausible - hunting and fishing are some of the easiest ways to obtain food after the apocalypse, and one of the few long-term solutions left. As cabins are one of the best places to live in, it shouldn't be surprising to find survivors there. 

It also goes along with my Free Merchants additions - hunters sell their surplus to Free Merchants, who in turn sell it to survivors who don't produce enough for themselves or their families. 

#### Describe the solution
1. I created a new hunter NPC who trades in food, water and furs. He has multiple dynamic speech options so it doesn't feel like you're always talking to the same person. 
2. NPCs will spawn in some cabins and campsites (I added two special types of these).

#### Describe alternatives you've considered
It would be great if, at some point in the future, they would wander around the area as well, but I have no idea how to code it.

#### Testing
I tested everything and it should work properly. 

EDIT: It seems that some checks failed due to reasons completely unrelated to any of my changes.

### CODE REVIEW

**Comment by davidpwbrown** [comment]
It would be better to make  a new faction for this hunter,
"wasteland_scavengers" is a legacy faction thing, where previously all dynamic NPCs and some static ones belonged to it, if you add your hunter to this faction, and someone kills this hunter, then some members of the refugee center will telepathically know and go hostile to the player.

Or use the existing no_faction faction, which has different rules in the code, ( i.e it dosnt telepathically communicate wrongdoing across its "members" )

File: data/json/mapgen/cabin.json [hunk 1]
```diff
@@ -127,6 +127,45 @@
       "place_monsters": [ { "monster": "GROUP_ZOMBIE", "x": [ 2, 21 ], "y": [ 2, 21 ] } ]
     }
   },
+  {
+    "type": "mapgen",
+    "method": "json",
+    "om_terrain": [ "cabin" ],
+    "weight": 333,
```
Comment (davidpwbrown) [inline data/json/mapgen/cabin.json]: I assume this weighting means that quite a few cabins will contain duplicates of this NPC ?

**Comment by Brytenwalda** [comment]
> 
> 
> It would be better to make a new faction for this hunter,
> "wasteland_scavengers" is a legacy faction thing, where previously all dynamic NPCs and some static ones belonged to it, if you add your hunter to this faction, and someone kills this hunter, then some members of the refugee center will telepathically know and go hostile to the player.
> 
> Or use the existing no_faction faction, which has different rules in the code, ( i.e it dosnt telepathically communicate wrongdoing across its "members" )

Changed it, they now have a faction of their own.

File: data/json/mapgen/campsite.json [hunk 2]
```diff
@@ -70,6 +70,67 @@
...
+        "#": "f_canvas_wall",
+        "'": "f_groundsheet",
+        "+": "f_canvas_door",
+        "-": "f_camp_chair",
+        "b": "f_firering",
+        "r": "f_tourist_table"
+      },
+      "place_items": [
+        { "item": "cannedfood", "x": [ 16, 4 ], "y": [ 21, 7 ], "chance": 75 },
+        { "item": "cannedfood", "x": [ 16, 4 ], "y": [ 21, 7 ], "chance": 75 },
```
Comment (Night-Pryanik) [inline data/json/mapgen/campsite.json]: This is a duplicate of a line above. Is it an error, or you just wanted to make `cannedfood` itemgroup have double rolls to spawn?

Also, `"y": [ 21, 7 ]` is a weird notation. Did you mean `"y": [ 7, 21 ]`?

File: data/json/npcs/NC_HUNTER.json [hunk 3]
```diff
@@ -67,23 +79,65 @@
     "id": "NC_HUNTER_misc",
     "items": [
       [ "sinew", 10 ],
-      [ "water_clean", 20 ],
-      [ "meat_cooked", 10 ],
+      [ "bolt_wood", 90 ],
+      [ "bolt_wood", 90 ],
```
Comment (Night-Pryanik) [inline data/json/npcs/NC_HUNTER.json]: You should use syntax like `{ "item": "bolt_wood", "prob": 90, "count": 2 }`. Here and below.

File: data/json/npcs/NC_HUNTER.json [hunk 4]
```diff
@@ -67,23 +79,65 @@
...
+    "type": "talk_topic",
+    "dynamic_line": [
+      "Hunting is becoming harder and harder these days.",
+      "Did you come here to trade?",
+      "Got anything to trade?",
+      "Good to see a friendly face.",
+      "Not many people left to trade with.",
+      "Hunting gives me hope that we can beat this.",
+      "One morning I shot an elk in my pajamas.  How he got into my pajamas I'll never know.",
+      { "u_male": [ "Sir.  Care to trade?" ], "no": [ "Ma'am.  Care to trade?." ] }
```
Comment (Night-Pryanik) [inline data/json/npcs/NC_HUNTER.json]:
```suggestion
      { "u_male": [ "Sir.  Care to trade?" ], "no": [ "Ma'am.  Care to trade?" ] }
```

File: data/json/mapgen/campsite.json (hunk 2 above)
Comment (Brytenwalda) [reply data/json/mapgen/campsite.json]: I wanted it to have double rolls for greater variety.

The notation is of course a mistake, corrected.

File: data/json/npcs/NC_HUNTER.json (hunk 3 above)
Comment (Brytenwalda) [reply data/json/npcs/NC_HUNTER.json]: Corrected

File: data/json/mapgen/campsite.json [hunk 5]
```diff
@@ -70,6 +70,67 @@
...
+        "#": "f_canvas_wall",
+        "'": "f_groundsheet",
+        "+": "f_canvas_door",
+        "-": "f_camp_chair",
+        "b": "f_firering",
+        "r": "f_tourist_table"
+      },
+      "place_items": [
+        { "item": "cannedfood", "x": [ 16, 4 ], "y": [ 7, 21 ], "chance": 75 },
+        { "item": "cannedfood", "x": [ 16, 4 ], "y": [ 7, 21 ], "chance": 75 },
```
Comment (Night-Pryanik) [inline data/json/mapgen/campsite.json]: Double rolls can be achieved by `{ "item": "cannedfood", "x": [ 16, 4 ], "y": [ 7, 21 ], "chance": 75, "repeat": 2 }`

File: data/json/mapgen/campsite.json [hunk 6]
```diff
@@ -70,6 +70,67 @@
...
+      "furniture": {
+        "#": "f_canvas_wall",
+        "'": "f_groundsheet",
+        "+": "f_canvas_door",
+        "-": "f_camp_chair",
+        "b": "f_firering",
+        "r": "f_tourist_table"
+      },
+      "place_items": [
+        { "item": "cannedfood", "x": [ 16, 4 ], "y": [ 7, 21 ], "chance": 75 },
```
Comment (Night-Pryanik) [inline data/json/mapgen/campsite.json]:
```suggestion
        { "item": "cannedfood", "x": [ 4, 16 ], "y": [ 7, 21 ], "chance": 75 },
```

File: data/json/mapgen/campsite.json [hunk 7]
```diff
@@ -70,6 +70,67 @@
...
+        { "item": "cannedfood", "x": [ 16, 4 ], "y": [ 7, 21 ], "chance": 75 },
+        { "item": "pasta", "x": [ 16, 4 ], "y": [ 7, 21 ], "chance": 75 },
```
Comment (Night-Pryanik) [inline data/json/mapgen/campsite.json]:
```suggestion
        { "item": "pasta", "x": [ 4, 16 ], "y": [ 7, 21 ], "chance": 75 },
```

File: data/json/mapgen/campsite.json [hunk 8]
```diff
@@ -70,6 +70,67 @@
...
+        { "item": "camping", "x": [ 10, 8 ], "y": [ 15, 13 ], "chance": 75 }
```
Comment (Night-Pryanik) [inline data/json/mapgen/campsite.json]:
```suggestion
        { "item": "camping", "x": [ 8, 10 ], "y": [ 13, 15 ], "chance": 75 }
```

**Comment by I-am-Erk** [comment]
@Brytenwalda I'd love to merge this and your other changes, they just need a few fixes and some conflict resolution. Feel free to reopen these if you come back to fix up the last bits. I can talk to you on discord to help fix the last bits if you need help.

**Pull request closed without merging**